core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -731,7 +731,13 @@ public PropServerConfiguration(
                 this.pgPendingWritersCacheCapacity = getInt(properties, env, PropertyKey.PG_PENDING_WRITERS_CACHE_CAPACITY, 16);
             }
 
-            this.walApplyWorkerCount = getInt(properties, env, PropertyKey.WAL_APPLY_WORKER_COUNT, cpuAvailable);
+            int walApplyWorkers = 2;
```

Comment:
Looks like fine tuning. :)
Why not just take the sum of configured input as a base - (ilp+pg+http = cpuUse) and take some % from it?

Reply:
`cpuUse` will be 0 or very small on most installations since it'll only stand for ILP writer threads and there are a few of them. Are you referring to `cpuAvailable`? What percentage do you have in mind?

Reply:
Both can be involved in the formula, but `cpuAvailable` doesn't say anything about the current configuration or load.
I'm thinking of a kind of "balancing" of the sides of the queue - the number of configured inputs on one side and the number of apply workers on the other or simply use all available `(cpuAvailable  - cpuUse )`. But ok, step function is acceptable here as well.

Reply:
> I'm thinking of a kind of "balancing" of the sides of the queue - the number of configured inputs on one side and the number of apply workers on the other

We have three potential pools on the left side: shared pool or pgwire pool or ILP I/O pool. We could use a certain percentage of minimal of them (say, 25%, but not less than 1 thread). WDYT?

This would be `cpuSpare` which is currently a step function.